curation-migration-datasync/dynamodb/curatedItemRecordModel.ts

```diff
@@ -66,6 +66,52 @@ export class CuratedItemRecordModel {
     }
   }
 
+  /**
+   * retrieves list of curatedItem by approvedItemExternalId
+   * Note: we have 1:N mapping of approvedItem externalId and curated_rec_id
+   * @param approvedItemExternalId approvedItem's externalId
+   */
+  public async getByApprovedItemExternalId(
+    approvedItemExternalId: string
+  ): Promise<CuratedItemRecord[]> {
+    const input: QueryCommandInput = {
+      TableName: config.aws.dynamoDB.curationMigrationTable,
+      IndexName: 'approvedItemExternalId-GSI',
+      KeyConditionExpression:
+        'approvedItemExternalId = :approvedItemExternalId',
+      ExpressionAttributeValues: {
+        ':approvedItemExternalId': approvedItemExternalId,
+      },
+    };
+    const res: QueryCommandOutput = await this.client.send(
+      new QueryCommand(input)
+    );
+
+    //this must always be false coz we are expecting only one item per record.
+    //if this gets thrown, we need to investigate the bug
+    if (res.LastEvaluatedKey) {
+      Sentry.captureMessage(
+        `method 'getByApprovedItemExternalId' called with '${approvedItemExternalId}'
+       has multiple pages of results that we are not handling!`
+      );
+    }
+
+    if (res.Items?.length) {
+      return res.Items.map((item): CuratedItemRecord => {
+        // force type safety
+        return {
+          curatedRecId: item.curatedRecId,
+          scheduledSurfaceGuid: item.scheduledSurfaceGuid,
+          scheduledItemExternalId: item.scheduledItemExternalId,
+          approvedItemExternalId: item.approvedItemExternalId,
+          lastUpdatedAt: item.lastUpdatedAt,
+        };
+      });
+    } else {
+      return [];
+    }
```

Comment:
I believe this should achieve the same thing
```suggestion
    return res.Items as CuratedItemRecord[]
```

Reply:
nice, thanks! changed for `getByApprovedItemExternalId` and `getByScheduledSurfaceGuid`.

 i tried doing the same for 
`getByCuratedId` and `getByScheduledItemExternalId`..but coz they need to be single record, we have to check for res.item? otherwise it throws an error.